tests/test_categorical_encoder.py

```diff
@@ -520,3 +521,28 @@ def test_RareLabelEncoder(dataframe_enc_big, dataframe_enc_big_na):
     df = pd.DataFrame(df)
     pd.testing.assert_frame_equal(X, df)
 
+
+def test_DecisionTreeCategoricalEncoder(dataframe_enc, dataframe_enc_rare, dataframe_enc_na):
+    # test case 1: user provides the categorical encoder
```

Comment:
could we separate each test case into a separate test as done here:
https://github.com/solegalli/feature_engine/blob/62d1ed19a6c32b6710516f262561cddca19ae7e9/tests/test_wrappers.py

It is better practice and it helps debugging later on. I know I do it as you did here, but I was advised to break i down :) so I am trying to incorporate better practices as we add more bits of code.